### ISSUE TITLE
[Retouche Module] mask oppacity unintuitive

### ISSUE THREAD

**Issue by AxelG-DE** [darktable-org/darktable#4019]
**Describe the bug**
In healing mode drawing a mask,  the mask opacity setting does not show up right away (shape selected non). You need to select a shape first.

**Especially when by chance a very low opacity is stored, that can be very confusing**

**To Reproduce**
1. Go to retouch module
2. Click on circular or elipse (in healing mode)
3. draw a mask to your picture
4. See that you cannot change the mask opacity

**Expected behavior**
The last drawn mask could be the selected shape and hence mask opacity would be visible.
(if really no shape selected, it could be greyed out, as nonchangeable, but dissapearing, that is not the best idea)

**Screenshots**
just after drawing a circle
![Screenshot_20200104_134858](https://user-images.githubusercontent.com/49004103/71765810-0bec6e80-2ef9-11ea-8db5-bcac6cf20a41.png)

and after activated the shape
![Screenshot_20200104_133940](https://user-images.githubusercontent.com/49004103/71765816-13ac1300-2ef9-11ea-8920-764038faf45f.png)

**Platform (please complete the following information):**
 - Darktable Version: 3.0.0 and current master
 - OS: Gentoo Linux
 - OpenCL activated 
 - 2x nvidia GTX1060, nvidia-drivers 440.11-r1

**Additional context**
 - Can you reproduce with another Darktable version? **YES**
 - Can you reproduce with a RAW or Jpeg or both? **RAW**
 - Are the steps above reproduce with a fresh edit? **YES**

**Comment by jade-nl** [comment]
I can confirm this.

You are able to set feather/opacity using the keyboard/mouse combo('s), but the mask opacity slider isn't visible in the module.

Only if you move the source (the tiniest bit) the slider shows up in the module.

**Comment by TurboGit** [comment]
I don't understand. The mask opacity can be changed while in creation mode with the ctrl+shift+scroll. Then when you create the first circle or ellipse you are still in creation mode. So you don't see the mask opacity indeed.

And yes the last mask is not selected when you exit the edit mode. Same with the free form BTW.

**Comment by jade-nl** [comment]
@TurboGit: Aren't you done creating after you right click? At that point I see a destination circle a source circle and an arrow between them.

I do _not_ see the mask opacity slider at the bottom of the retouch module. Only after clicking inside the source circle does the slider appear.

And like I stated before: Keyboard + mouse actions _do_ work.

**Comment by TurboGit** [comment]
Right, no opacity slider as there is no selected shape.

For me the mouse action to change opacity is working fine.

**Comment by jade-nl** [comment]
@TurboGit: *the penny drops* I think the English saying goes.....

Still: It is rather strange that you are able to change the opacity with the keyboard and mouse even though nothing is (seems to be?) selected. The shape is either selected (and shown as such) or it isn't. If it is then both ways often altering the opacity should be accessible if it isn't then neither should work.

As stated in OP's title: Not very intuitive.

**Comment by TurboGit** [comment]
@jade-nl : No I'm changing BEFORE placing the form in creation mode.

After that I need to select the proper shape to change the opacity of the corresponding shape.

**Comment by AxelG-DE** [comment]
@TurboGit What you describe is what I'd call unituitive :-) as in that moment, there is no indication, the opacity could be changed :-) 
Please understand me right, I am a keyboard-freak too, however, in that specific scenario, I got trapped already several times, as I could not see, what is going on in the first place, particularly as the opacity is stored from previous setup, which is basically a good thing, just comes tricky here...

**Comment by AxelG-DE** [comment]
Actually I just figured out, in the **exposure module** it is even **worse**.
using a circlular drawn mask the opacity is always 0 (I am not talking about refinement part) and **need to be adjusted** by ctrl+mouse **every time** I set a cirle and there is **no** this **white shaded indicator** as I have it for brush....

**Comment by junkyardsparkle** [comment]
> I must correct my self, it get stored, if one changes it (ctrl-mouse) prior left clicking

Hah, thanks for pointing that out, I was still trying to figure out wtf was going on there... I think there might be some assumptions going on here about how people use these tools which aren't always true?

> BUT at that time also no indication, you're working blind in case of circle (current master and 3.0.0)

To me, the interesting thing about #3740 is the inclusion of visible controls for this stuff... I think it would prevent many headaches. :-)

**Comment by TurboGit** [comment]
> BUT at that time also **no indication, you're working blind** in case of circle (current master and 3.0.0)

Not completely true, if the top panel is visible you have hints about the opacity.

As @junkyardsparkle said, it will be better when the redesign for hint is done to ensure this information is always visible.

**Comment by aurelienpierre** [comment]
Duplicate of #3740.
